### PR TITLE
(CDAP-13143) Supports recursive Schema correctly

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT6332-3

File: cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java [hunk 1]
```diff
@@ -373,6 +373,98 @@ public void testSerializable() throws IOException, ClassNotFoundException {
...
+    // Test multi-levels recursion. It has the following structure:
+    // Parent {
+    //   List<Child> children
+    // }
+    // Child {
+    //   GrantChild child
+    // }
+    // GrantChild {
+    //   Child parent
+    //   Parent grantParent
```
Comment (sagarkapare) [inline cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java]: nit: you mean `grandChild` and `grandParent`?

Comment (chtyim) [reply cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java]: No. The parent of the `GrantChild` type is `Child`, and the grant parent of the `GrantChild` is `Parent`, as what being constructed in the Schema below

Comment (chtyim) [reply cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java]: Fixed.

**Comment by sagarkapare** [comment]
LGTM 👍

**Pull request merged**